workflow-to-js-transformer/src/main/resources/Workflow.xslt

```diff
@@ -266,25 +272,61 @@
+                actionDetails();
+                recordActionCompletedOrFailed(document);
+                return ALREADY_EXECUTED;
+            }
+            // propagate the post-processing field to response custom data if it exists on the document custom data
+            var responseCustomData = actionDetails.customData ? actionDetails.customData : {};
+            if(!isEmpty(document.getCustomData(postProcessingScriptPropertyName))){
+                responseCustomData[postProcessingScriptPropertyName] = document.getCustomData(postProcessingScriptPropertyName);
+            }
 
-        // propagate the post-processing field to response custom data if it exists on the document custom data
-        var responseCustomData = actionDetails.customData ? actionDetails.customData : {};
-        if(!isEmpty(document.getCustomData(postProcessingScriptPropertyName))){
-        responseCustomData[postProcessingScriptPropertyName] = document.getCustomData(postProcessingScriptPropertyName);
+            // Update document destination queue to that specified by action and pass appropriate settings and customData
+            var queueToSet = !isEmpty(actionDetails.queueName) ? actionDetails.queueName : actionDetails.workerName+"Input";
+            var responseOptions = document.getTask().getResponseOptions();
+            responseOptions.setQueueName(queueToSet);
+            responseOptions.setCustomData(responseCustomData);
+            return ACTION_TO_EXECUTE;
         }
 
-        // Update document destination queue to that specified by action and pass appropriate settings and customData
-        var queueToSet = !isEmpty(actionDetails.queueName) ? actionDetails.queueName : actionDetails.workerName+"Input";
-        var responseOptions = document.getTask().getResponseOptions();
-        responseOptions.setQueueName(queueToSet);
-        responseOptions.setCustomData(responseCustomData);
+        // executes the field mapping action on the document
+        function executeFieldMapping(document, mappings){
+        // get the field values to map (from the document)
+        var documentFieldsValuesMap = {};
+        for(var mappingKey in mappings) {
+        var documentFieldToMapFrom =  document.getField(mappingKey);
+        if(!documentFieldToMapFrom.hasValues()){
+        continue;
+        }
+        documentFieldsValuesMap[mappingKey] = documentFieldToMapFrom.getValues();
+        documentFieldToMapFrom.clear();
         }
 
-    </xsl:template>
+        // for each mapping add the original field value with the new key
+        for(var mappingKey in mappings) {
+        var mappingDestination = mappings[mappingKey];
+        var documentFieldToMapTo = document.getField(mappingDestination);
+        for each(var fieldValue in documentFieldsValuesMap[mappingKey]) {
+        if(fieldValue.isReference()){
+        documentFieldToMapTo.addReference(fieldValue.getReference());
+        continue;
```

Comment:
Using `else` instead of `continue` would be cleaner.

Reply:
Changed to an else.